griptape/drivers/observability/griptape_cloud_observability_driver.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import annotations
+
+import os
+from typing import TYPE_CHECKING, Optional
+from urllib.parse import urljoin
+from uuid import UUID
+
+import requests
+from attrs import Attribute, Factory, define, field
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor, SpanExporter, SpanExportResult
+from opentelemetry.sdk.util import ns_to_iso_str
+from opentelemetry.trace import INVALID_SPAN, get_current_span
+
+from griptape.drivers.observability.open_telemetry_observability_driver import OpenTelemetryObservabilityDriver
+
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+
+    from opentelemetry.sdk.trace import ReadableSpan, SpanProcessor
+
+
+@define
+class GriptapeCloudObservabilityDriver(OpenTelemetryObservabilityDriver):
+    service_name: str = field(default="griptape-cloud", kw_only=True)
+    base_url: str = field(
+        default=Factory(lambda: os.getenv("GT_CLOUD_BASE_URL", "https://cloud.griptape.ai")), kw_only=True
+    )
+    api_key: str = field(default=Factory(lambda: os.getenv("GT_CLOUD_API_KEY")), kw_only=True)
```

Comment:
Maybe out of the scope for this PR, but I just realized `getenv` returns `str | None` which is at odds with `api_key: str`.

Should probably replace with `os.environ` or maybe a griptape wrapper, similar to `import_optional_dependency`.

Reply:
This did cross my mind, but decided to just go with the flow. I can do something about it

Reply:
Actually, since this pattern touches files outside of the current PR, we can fast follow. I'll create a task.